bool action
----
bool is_open